code/missionui/missiondebrief.cpp

```diff
@@ -952,10 +952,11 @@ void debrief_choose_voice(char *voice_dest, size_t buf_size, char *voice_base, i
 void debrief_choose_medal_variant(char *buf, int medal_earned, int zero_based_version_index)
 {
 	Assert(buf != NULL && medal_earned >= 0 && zero_based_version_index >= 0);
+	Assertion(strlen(Resolution_prefixes[gr_screen.res]) + strlen(Medals[medal_earned].debrief_bitmap) < MAX_FILENAME_LEN, "The filename for %s is over the filename length limit of 31 characters.  On release builds, this filename will be shortened and will not work.", buf);
```

Comment:
Use `%d` instead of writing the number of characters explicitly.  Also the filename is almost certainly truncated on both release _and_ debug.

Reply:
It trips an assert on debug, so you're not going to be able to continue, and the user already knows this.  I am encouraging the modder not to ignore this, because it will not do what they want it to do on release.

Reply:
Mmm, ok